File: InternetForumBackend/API/Data/Models/UserPostVote.cs
namespace API.Data.Models;

public class UserPostVote
{
    public int UserId { get; set; }
    public User User { get; set; }

    public int PostVoteId { get; set; }

    public PostVote PostVote { get; set; }

    //upvote/downvote
    public int PostVoteDirection { get; set; }
}

File: InternetForumFrontend/src/app/Models/UserPostVote.ts
import { PostVote } from "./PostVote";
import { User } from "./User";

// NOTE: ? = null, ! = not null
export class UserPostVote {
        userId?: number;
        user?: User;
        postVote?: PostVote;
        postVoteId?: number;
        postVoteDirection!: number;
}